tests/test_TFNetworkLayer.py

```diff
@@ -981,33 +981,53 @@ def test_SplitDimsLayer_resolve_dims():
   assert_equal(SplitDimsLayer._resolve_dims(old_dim=2 * 3 * 5, new_dims=(2, 3, -1, 1)), (2, 3, 5, 1))
 
 
-def _check_MergeDimsLayer(session, in_data_opts, in_static_shape, opts, out_data_shape, out_static_shape):
+def _check_MergeDimsLayer(session, in_data_opts, in_static_shape, opts, out_data_shape, out_static_shape,
+                          out_size_placeholder=None):
   """
   :param TFCompat.v1.Session session:
   :param dict[str] in_data_opts:
   :param tuple[int] in_static_shape:
   :param dict[str] opts: for MergeDimsLayer
   :param tuple[int|None] out_data_shape:
   :param tuple[int] out_static_shape:
+  :param dict[int, tuple[int]] out_size_placeholder:
   :rtype: MergeDimsLayer
   """
   net = TFNetwork(extern_data=ExternData())
   rnd = numpy.random.RandomState(42)
   src = InternalLayer(name="src", network=net, out_type=in_data_opts)
   print("input:", src.output)
   src.output.placeholder = tf.constant(rnd.normal(size=in_static_shape).astype("float32"), dtype=tf.float32)
-  src.output.size_placeholder = {}  # not sure if enough...
+  src.output.size_placeholder = {}
+
+  if out_size_placeholder:
+    n_batch = in_static_shape[src.output.batch_dim_axis]
+    for axis, dim in enumerate(in_data_opts['shape']):
```

Comment:
Why not use `src.output.batch_shape` instead? That would be cleaner.

Reply:
(Also, by convention, I usually call `axis_wo_b` the var if it does not contain the batch-dim, and otherwise simply `axis`, which implies that it is counted with batch-dim.)

Reply:
```suggestion
    for axis, dim in enumerate(src.output.batch_shape):
      axis_wo_b = src.output.get_batch_axis_excluding_batch(axis)
```